mla/injectors.py

```diff
@@ -200,20 +200,421 @@ def inject_signal_events(self, reduced_sim: np.ndarray) -> np.ndarray:
+        """Function info...
+        
+        Prunes the simulation set to only events close to a given source and calculate the
+        weight for each event. Adds the weights as a new column to the simulation set.
+            
+        Args:
+            reduced_sim: simulation set.
+            flux_norm: A flux normaliization to adjust weights.
+            gamma: A spectral index to adjust weights.
+            spectrum: spectral.BaseSpectrum object.Alternative for user not passing flux_norm and gamma. 
+            sampling_width: The bandwidth around the source declination to cut events.
+        
+        Returns:
+            A reweighted simulation set around the source declination.
+        """
+        if livetime is None:
+            try:
+                livetime = self.signal_time_profile.effective_exposure()
+            except:
+                print("no livetime and no signal time profile in injector.Using 1 day to weight the MC")
+                livetime = 1
+        if spectrum is not None:
+            reduced_sim['weight'] = reduced_sim['ow'] * spectrum(reduced_sim['trueE']) * livetime * 3600 * 24
+        else:
+            reduced_sim['weight'] = reduced_sim['ow'] * flux_norm * (reduced_sim['trueE']/100.e3)**gamma * livetime * 3600 * 24
 
-    def __init__(self, source: Dict[str, float],
-                 bg_time_profile: time_profiles.GenericProfile,
-                 sig_time_profile: time_profiles.GenericProfile) -> None:
-        """Docstring"""
-        super().__init__(source)
-        self.bg_time_profile = bg_time_profile
-        self.sig_time_profile = sig_time_profile
+        return reduced_sim
+        
+    def inject_background_events_No_time(self, event_model: models.EventModel) -> np.ndarray:
+        """Function info...
+        
+        More function info...
+        
+        Returns:
+            An array of injected background events.
+        """
+        # Get the number of events we see from these runs
+        n_background = event_model.grl['events'].sum()
+        n_background_observed = np.random.poisson(n_background)
+
+        # How many events should we add in? This will now be based on the
+        # total number of events actually observed during these runs
+        background = np.random.choice(event_model.data, n_background_observed).copy()
+
+        # Randomize the background RA
+        background['ra'] = np.random.uniform(0, 2*np.pi, len(background))
+        
+        return background
 
     def inject_background_events(self,
-                                 event_model: models.EventModel) -> np.ndarray:
-        """Docstring"""
-        raise NotImplementedError
+                                 event_model: models.EventModel,
+                                 background_time_profile: Optional[time_profiles.GenericProfile] = None,
+                                 background_window: Optional[float] = 14
+                                 ) -> np.ndarray:
```

Comment:
I think it's better to have the time profiles as attributes and not pass them every time